src/main/java/org/tron/common/runtime/Runtime.java

```diff
@@ -521,32 +523,31 @@ public void go() {
             runtimeError = "REVERT opcode executed";
           }
         } else {
-          getResult().getInternalTransactions().add(0,rootInternalTransaction);
           deposit.commit();
         }
       } else {
-        if (!trxType.equals(TRX_PRECOMPILED_TYPE)) {
-          getResult().getInternalTransactions().add(0,rootInternalTransaction);
-        }
         deposit.commit();
       }
     } catch (JVMStackOverFlowException e) {
       program.spendAllEnergy();
       result = program.getResult();
       result.setException(e);
+      result.rejectInternalTransactions();
       runtimeError = result.getException().getMessage();
       logger.info("JVMStackOverFlowException: {}", result.getException().getMessage());
     } catch (OutOfResourceException e) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this left curly brace to the beginning of next line of code. ([squid:LeftCurlyBraceStartLineCheck](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1633&issues=AWak_C2wdPPRVdwyhXMV&open=AWak_C2wdPPRVdwyhXMV)

[ISSUE_KEY]: AWak_C2wdPPRVdwyhXMV